src/h2n_generate.erl

```diff
@@ -0,0 +1,241 @@
+    blank_line(follow(text([Name, " ="]), text([NixName, ";"])));
+format_position(_, _, _) ->
+    empty().
+
+-spec app_body(h2n_fetcher:dep_despc(), [binary()]) -> prettypr:document().
+app_body(Dep = #dep_desc{app = {Name, Vsn},
+                         sha = Sha},
+         Deps) ->
+    erlang_deps(Deps),
+    par([break(sep([text("buildHex"), text("{")]))
+        , nest(par([key_value(<<"name">>, Name)
+                   , key_value(<<"version">>, Vsn)
+                   , key_value(<<"sha256">>, Sha)
+                   , format_compile_port(Dep)
+                   , erlang_deps(Deps)
+                   , meta(Dep)]))
+        , text("}")]).
+
+-spec format_compile_port(h2n_fetcher:dep_desc()) -> prettypr:document().
+format_compile_port(#dep_desc{has_native_code = true}) ->
+    break(follow(text(["compilePort", " ="]), text("true;")));
+format_compile_port(_) ->
+    empty().
+
+-spec meta(h2n_fetcher:dep_despc()) -> prettypr:document().
+meta(#dep_desc{description = Description
+              , licenses = Licenses
+              , homepage = Homepage}) ->
+    format_licenses(Licenses),
+    par([sep([text("meta")
+             , text("=")
+             , text("{")])
+        , nest(par([format_description(Description)
+                   , format_licenses(Licenses)
+                   , case Homepage of
+                         no_source -> empty();
+                         _ -> key_value(<<"homepage">>, Homepage)
+                     end]))
+        , text("};")]).
+
+-spec format_description(binary()) -> prettypr:document().
+format_description(<<"">>) ->
+    empty();
+format_description(Description)
+  when erlang:size(Description) =< 80 ->
+    key_value(<<"description">>, Description);
+format_description(Description) ->
+    DescBody = erlang:iolist_to_binary(["''", Description, "'';"]),
+    break(follow(text(["longDescription", " ="])
+                , text_par(erlang:binary_to_list(DescBody)))).
+
+
+-spec format_licenses([binary()]) -> prettypr:document().
+format_licenses([]) ->
+    empty();
+format_licenses([License]) ->
+    sep([text("license"), text("="), text(["stdenv.lib.licenses.", License, ";"])]);
+format_licenses(Values) ->
+    sep([text("license"), text("="), text("with")
+        , text("stdenv.lib.licenses;")
+        , text("[")
+        , nest(expand_arg_list(Values, "", []))
+        , text("];")]).
+
+-spec erlang_deps([binary()]) -> prettypr:document().
+erlang_deps([]) ->
+    empty();
+erlang_deps(Deps) ->
+    blank_line(follow(text("erlangDeps =")
+                     , par([text("[")
+                           , nest(expand_arg_list(Deps, "", []))
+                           , text("];")]))).
+
+-spec key_value(binary(), binary()) -> prettypr:document().
+key_value(Key, Value) ->
+    break(follow(text([Key, " ="]), text(["\"", Value, "\";"]))).
+
+-spec section_header([binary()]) -> prettypr:document().
+section_header(Deps) ->
+    sep([text("{ ")
```

Comment:
Oh, I just realized that I we probably misunderstood each other. I meant a double space after `{` and before `buildHex`, which should be single space, as seen here:

https://github.com/ericbmerritt/nixpkgs/blob/f5baf1f5c721b434a87a5f6f02f64b8b39863d7c/pkgs/development/erlang-modules/hex-packages.nix#L110